sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -138,15 +138,13 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
             s"$cmd expects a view.")
 
       case u: UnresolvedView =>
-        u.failAnalysis(s"View not found: ${u.multipartIdentifier.quoted}")
+        throw new NoSuchTableException(u.multipartIdentifier)
 
       case u: UnresolvedTableOrView =>
-        val viewStr = if (u.allowTempView) "view" else "permanent view"
-        u.failAnalysis(
-          s"Table or $viewStr not found: ${u.multipartIdentifier.quoted}")
+        throw new NoSuchTableException(u.multipartIdentifier)
```

Comment:
```suggestion
        u.failAnalysis(errorClass = "TABLE_OR_VIEW_NOT_FOUND",
          messageParameters =
            Array(u.multipartIdentifier.map(part => quoteIdentifier(part)).mkString(".")))
```

Reply:
In this way, the origin of `u` will be passed to the exception.

Reply:
Also, I think I need to keep the origin of Analysis exception if it is present in https://github.com/apache/spark/pull/37841/files#diff-ed19f376a63eba52eea59ca71f3355d4495fad4fad4db9a3324aade0d4986a47R213 
I will create a PR for the fix.